### PR TITLE
Add support for CMake (buildsystem generator)

### PR DESCRIPTION
**Pull request by pajlada**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

# Known issues
 - [x] Mac scaling is weird with cmake
 - [x] Mac icon doesn't work

### CODE REVIEW

**Comment by pajlada** [comment]
TODO: 
MacOS build failing because PajladaSettings doesn't build with our version of the FindRapidJSON package

**Comment by KararTY** [comment]
CMAKE build for windows missing libcrypto & libssl .dll files:
image
image

Copying them over from existing nightly build allows Chatterino CMAKE build to run, however.

**Comment by gempir** [comment]
MacOS is missing the Icon 
image

File: conanfile.txt [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 [generators]
 qmake
+cmake
 
 [options]
 openssl:shared=True
 
 [imports]
-bin, *.dll -> ./Chatterino2 @ keep_path=False
+bin, *.dll -> ./bin @ keep_path=False
```
Comment (AnotherFoxGuy) [inline conanfile.txt:13]: This should fix the missing .dll files on windows
```suggestion
bin, *.dll -> ./bin @ keep_path=False
bin, *.dll -> ./Chatterino2 @ keep_path=False
```

**Comment by pajlada** [comment]
> CMAKE build for windows missing libcrypto & libssl .dll files:
> image
> image
> 
> Copying them over from existing nightly build allows Chatterino CMAKE build to run, however.

Could you try the new windows zip in the cmake-build release https://github.com/Chatterino/chatterino2/releases/tag/cmake-build ?

**Comment by KararTY** [comment]
> Could you try the new windows zip in the cmake-build release https://github.com/Chatterino/chatterino2/releases/tag/cmake-build ?

The missing .dll files are now included and it launches now. Cool.

**Pull request merged**